src/main/java/com/artipie/asto/google/GoogleStorage.java

```diff
@@ -0,0 +1,112 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.google;
+
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.Transaction;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.StorageOptions;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Storage that holds data in Google storage.
+ *
+ * @since 0.1
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class GoogleStorage implements Storage {
+
+    /**
+     * Google storage client.
+     */
+    private final com.google.cloud.storage.Storage client;
+
+    /**
+     * Bucket name.
+     */
+    private final String bucket;
+
+    /**
+     * Ctor.
+     *
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final String bucket) {
+        this(StorageOptions.getDefaultInstance(), bucket);
+    }
+
+    /**
+     * Ctor.
+     * @param options Options for Google storage client
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final StorageOptions options, final String bucket) {
+        this.client = options.getService();
+        this.bucket = bucket;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(final Key key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Collection<Key>> list(final Key prefix) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> save(final Key key, final Content content) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> move(final Key source, final Key destination) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Content> value(final Key key) {
```

Comment:
@Guseyn can we test this method?

Reply:
@Vatavuk I am going to rewrite it. I am not sure how we can test this as for that stuff I need test account in GSP and I cannot find mocks for google storage. Maybe it's possible to write my own mocks but I would do it as a separate task if you don't mind because it can be complex thing to do.

Reply:
@Guseyn ok